webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DatasetDeleter.scala

```diff
@@ -31,23 +41,264 @@ trait DatasetDeleter extends LazyLogging with DirectoryConstants {
+        // Two cases exist here: 1. The layer is a regular directory where each mag is a symlink
+        // 2. The layer is a symlink to the other layer itself.
+        // We can handle both by deleting the layer and creating a new symlink.
+        Files.delete(linkedLayerPath)
+        logger.info(
+          s"Deleting existing symlink at $linkedLayerPath linking to $sourceDataSource/$sourceLayer, creating new symlink")
+        Files.createSymbolicLink(linkedLayerPath, relativizeSymlinkPath(targetPath, linkedLayerPath))
+      } else {
+        if (!Files.exists(linkedLayerPath)) {
+          // This happens when the layer is a locally explored dataset, where the path is directly written into the properties
+          // and no layer directory actually exists.
+          Files.createSymbolicLink(linkedLayerPath, relativizeSymlinkPath(targetPath, linkedLayerPath))
+        } else {
+          // This should not happen, since we got the info from WK that a layer exists here
+          logger.warn(s"Trying to recreate symlink at layer $linkedLayerPath, but it does not exist!")
+        }
+      }
+    }
+
+    // For every mag that linked to this layer, we need to update the symlink
+    // We need to discard the already handled mags (fully linked layers)
+
+    layerMags.foreach { magLinkInfo =>
+      val mag = magLinkInfo.mag
+      val newMagPath = targetPath.resolve(mag.mag.toMagLiteral(true))
+      magLinkInfo.linkedMags
+        .filter(linkedMag => !fullLayerLinks.contains((linkedMag.dataSourceId, linkedMag.dataLayerName))) // Filter out mags that are fully linked layers, we already handled them
+        .foreach { linkedMag =>
+          updateMagSymlinks(newMagPath, linkedMag)
+        }
+    }
+
+  }
+
+  private def handleLayerSymlinks(dataSourceId: DataSourceId,
+                                  layerName: String,
+                                  linkedMags: List[MagLinkInfo]): Box[Unit] =
+    tryo {
+      val fullyLinkedLayersOpt = getFullyLinkedLayers(linkedMags)
+      fullyLinkedLayersOpt match {
+        case Some(fullLayerLinks) =>
+          moveLayer(dataSourceId, layerName, fullLayerLinks, linkedMags)
+        case None =>
+          logger.info(s"Found incomplete symlinks to layer; Moving mags from $dataSourceId to other datasets")
+          linkedMags.foreach { magLinkInfo =>
+            val magToDelete = magLinkInfo.mag
+            if (magLinkInfo.linkedMags.nonEmpty) {
+              if (magToDelete.hasLocalData) {
+                // Move mag to a different dataset
+                val magPath = dataBaseDir
+                  .resolve(dataSourceId.organizationId)
+                  .resolve(dataSourceId.directoryName)
+                  .resolve(layerName)
+                  .resolve(magToDelete.mag.toMagLiteral(true))
+                // Select an arbitrary linked mag to move to
+                val target = magLinkInfo.linkedMags.head
```

Comment:
another `.head` which would lead to non-readable exception. maybe add nonEmpty assertion